File: Second Year/PDOO/Napakalaki/src/NapakalakiGame/Prize.java
/*  
 *  Programación y diseño orientado a objetos
 *  Grado en Ingeniería Informática
 * 
 *  
 *
 *  Angela Castro Jiménez
 *  Francisco Carrillo Pérez
 */
package NapakalakiGame;


public class Prize {
    
    private int level;
    private int treasures;
    
    public Prize (){
      level = 0;
      treasures =0; 
    }
    
    public Prize (int t, int l) {
        treasures = t;
        level = l;
    }
    
    public int getTreasures(){
        return treasures;
    }
    
    public int getLevels(){
        return level;
    }
    
    public String toString(){
        return "Tesoros = "+Integer.toString(treasures)+". Niveles = " + Integer.toString(level)+".";   
    }

    
   /* PRUEBA 
        public static void main(String args[]){
           Prize premio = new Prize (2,4);
           System.out.println(premio.GetLevel());
           System.out.println(premio.GetTreasures()); 
           System.out.println(premio.toString());
        } */
}

File: Second Year/PDOO/Napakalakirb/lib/Prize.rb
# encoding: UTF-8
#
#   Programación y diseño orientado a objetos
#   Grado en Ingeniería Informática
#
#  Ángela Castro Jimenez
#  Francisco Carrillo Pérez
require_relative 'Monster'
require_relative 'BadConsequence'
module Napakalaki
  class Prize
    attr_accessor :level, :treasures 
  
    def initialize(level,treasures)
      @level = level
      @treasures = treasures
    end

    def self.Prize(l,t)
      new(l,t)
    end
    def getLevel
      return @level
    end
    def getTreasures
      return @treasures
    end
    def to_s 
      "\nTesoros ganados: #{@treasures} \n Niveles ganados: #{@level}"
    end
 
  end
end

